src/launch/config.c

```diff
@@ -781,6 +790,8 @@ static void config_parser_begin_fn(void *userdata, const XML_Char *name, const X
                 if (r)
                         goto failed;
 
+                c_list_link_tail(&state->root->servicedir_list, &node->servicedir_link);
+
```

Comment:
I don't think this works. We don't parse configurations in linear order. The only guarantee you get is that the node-list *afterwards* is in linear order.

For example, we always parse a single file *fully* before parsing any included files. But we make sure when parsing the included files, we insert them at the correct position in the node-tree/list. But if you now keep a separate order in `servicedir_list`, based on the time we parse nodes, you will get out-of-sync.

I think the only valid solution is to do a post-processing step. Once all the nodes have been parsed, go over them *once more* linearly, to get further links (like the servicedir list).